src/pages/ClientMain.vue

```diff
@@ -8,46 +8,64 @@
         <Map/>
         <Contacts/>
     </div>
+    <div v-else><CircleSpinner /></div>
 </template>
 
 <script>
-    import MainPart from '@/components/ClientMainParts/MainPart';
-    import DetailsPart from "@/components/ClientMainParts/DetailsPart";
-    import PreacherPart from '@/components/ClientMainParts/PreacherPart';
-    import VideoInvite from "@/components/ClientMainParts/VideoInvite";
-    import Registration from "@/components/Registration";
-    import Map from "@/components/ClientMainParts/Map";
-    import Contacts from "@/components/ClientMainParts/Contacts";
+import { mapState, mapActions } from 'vuex';
+import MainPart from '@/components/ClientMainParts/MainPart';
+import DetailsPart from '@/components/ClientMainParts/DetailsPart';
+import PreacherPart from '@/components/ClientMainParts/PreacherPart';
+import VideoInvite from '@/components/ClientMainParts/VideoInvite';
+import Registration from '@/components/Registration';
+import Map from '@/components/ClientMainParts/Map';
+import Contacts from '@/components/ClientMainParts/Contacts';
 
-    import { mapState, mapActions } from 'vuex';
-
-      export default {
-        name: "ClientMain",
-        components: {
-          MainPart,
-          DetailsPart,
-          PreacherPart,
-          VideoInvite,
-          Registration,
-          Map,
-          Contacts,
-        },
-        computed: mapState({
-          seminar: state => state.seminars.seminar,
-        }),
-        methods: {
-          ...mapActions('seminars', [
-            'fetchCurrentSeminar',
-          ]),
-          ...mapActions('preachers', [
-            'fetchCurrentPreacher',
-          ]),
-        },
-        created() {
-          this.fetchCurrentSeminar();
-          this.fetchCurrentPreacher(this.seminar.preacher_id);
-        }
-      }
+export default {
+  name: 'ClientMain',
+  components: {
+    MainPart,
+    DetailsPart,
+    PreacherPart,
+    VideoInvite,
+    Registration,
+    Map,
+    Contacts,
+  },
+  data() {
+    return {
+      monthNumber: (new Date().getMonth() + 1),
+    };
+  },
+  computed: {
+    ...mapState('seminars', {
+      seminarsLoading: (state) => state.loading,
+      seminar: 'seminar',
+    }),
+    ...mapState('preachers', {
+      preacherLoading: (state) => state.loading,
+    }),
+    ...mapState('lessons', {
+      lessonsLoading: (state) => state.loading,
+    }),
+  },
+  methods: {
+    ...mapActions('seminars', [
+      'fetchCurrentSeminar',
+    ]),
+    ...mapActions('preachers', [
+      'fetchCurrentPreacher',
+    ]),
+    ...mapActions('lessons', [
+      'fetchLessonsByMonth',
+    ]),
+  },
+  async created() {
+    await this.fetchCurrentSeminar();
+    await this.fetchCurrentPreacher(this.seminar.preacher_id);
+    await this.fetchLessonsByMonth(this.monthNumber);
```

Comment:
for which purposes this await had been used?

Reply:
For calling method fetchCurrentPreacher, for example, we need preacher_id from seminar, but we cannot use it, because seminar is the result of the previous method.
By the way, the last await is redundant.